### PR TITLE
[PAPI-30] [PAPI-238] Liquidity pool controller validation & examples

### PR DESCRIPTION
**Pull request by drmathias**
* Request validation
* Added examples to liquidity pool requests/responses
* Documented liquidity pool responses
* Organised request models

### CODE REVIEW

File: src/Opdex.Platform.WebApi/Controllers/LiquidityPoolsController.cs [hunk 1]
```diff
@@ -43,8 +41,14 @@ public LiquidityPoolsController(IMediator mediator, IMapper mapper, IApplication
     /// <param name="filters">Liquidity pool filter options.</param>
     /// <param name="cancellationToken">Cancellation token.</param>
     /// <returns><see cref="LiquidityPoolsResponseModel"/> of matching results and paging details.</returns>
+    /// <response code="200">Liquidity pool results returned.</response>
+    /// <response code="400">The request is not valid.</response>
+    /// <response code="401">Unauthorized.</response>
+    /// <response code="429">Too many requests.</response>
```
Comment (mrtpain) [inline src/Opdex.Platform.WebApi/Controllers/LiquidityPoolsController.cs:47]: Just a thought, wonder if should add 500 to all endpoints with 'Unexpected Error', not for this PR really just in general.

File: src/Opdex.Platform.WebApi/Controllers/LiquidityPoolsController.cs [hunk 2]
```diff
@@ -77,9 +87,16 @@ public LiquidityPoolsController(IMediator mediator, IMapper mapper, IApplication
...
+    /// <response code="200">Liquidity pool details found.</response>
+    /// <response code="400">The request is not valid.</response>
+    /// <response code="401">Unauthorized.</response>
+    /// <response code="404">Liquidity pool not found.</response>
+    /// <response code="429">Too many requests.</response>
     [HttpGet("{address}")]
     [ProducesResponseType(typeof(LiquidityPoolResponseModel), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(ValidationProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status401Unauthorized)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
```
Comment (mrtpain) [inline src/Opdex.Platform.WebApi/Controllers/LiquidityPoolsController.cs:100]: can add typeof(ProblemDetails)

**Review by mrtpain — APPROVED** [review]

File: src/Opdex.Platform.WebApi/Controllers/LiquidityPoolsController.cs (hunk 1 above)
Comment (drmathias) [reply src/Opdex.Platform.WebApi/Controllers/LiquidityPoolsController.cs:47]: I'd say so, I'll add them

Comment (drmathias) [reply src/Opdex.Platform.WebApi/Controllers/LiquidityPoolsController.cs]: I've added a global operation processor which will apply 500 response documentation to all endpoints. Replaced 429 documentation with a global operation processor as well.

**Pull request merged**